app/src/main/java/com/cookandroid/week1/view/activity/LoginActivity.kt

```diff
@@ -0,0 +1,75 @@
+package com.cookandroid.week1.view.activity
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModelProvider
+import com.cookandroid.week1.R
+import com.cookandroid.week1.databinding.ActivityLoginBinding
+import com.cookandroid.week1.view.model.LoginViewModel
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+
+    private val loginLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            result.data?.let { data ->
+                val id = data.getStringExtra(getString(R.string.id))
+                val nickname = data.getStringExtra(getString(R.string.nickname))
+                val address = data.getStringExtra(getString(R.string.address))
+
+                val intent = Intent(this, MainActivity::class.java).apply {
+                    putExtra(getString(R.string.id), id)
+                    putExtra(getString(R.string.nickname), nickname)
+                    putExtra(getString(R.string.address), address)
+                }
+
+                startActivity(intent)
```

Comment:
```suggestion
                Intent(this, MainActivity::class.java).apply {
                    putExtra(getString(R.string.id), id)
                    putExtra(getString(R.string.nickname), nickname)
                    putExtra(getString(R.string.address), address)
                    startActivity(this)
                }
```
처럼 작성하면 intent 객체 생성 없이 진행할 수도 있답니닷